src/coreclr/gc/gc.cpp

```diff
@@ -30401,7 +30410,12 @@ void gc_heap::plan_phase (int condemned_gen_number)
             }
 #endif //FEATURE_EVENT_TRACE
         }
-#ifndef USE_REGIONS
+#ifdef USE_REGIONS
+        if (!computed_end_space_p)
+        {
+            get_gen0_end_plan_space();
```

Comment:
Let combine the call to compute end space for both sweep and compact to after the GC join